packages/taste-test/src/TasteTest.class.ts

```diff
@@ -0,0 +1,234 @@
+import { AssetAmount } from "@sundaeswap/asset";
+import type {
+  ITxBuilder,
+  ITxBuilderFees,
+  ITxBuilderReferralFee,
+} from "@sundaeswap/core";
+import {
+  Data,
+  Datum,
+  SpendingValidator,
+  Tx,
+  TxComplete,
+  UTxO,
+  fromText,
+  toUnit,
+  type Lucid,
+  type MintingPolicy,
+} from "lucid-cardano";
+
+import { IDepositArgs } from "./@types";
+import {
+  DiscoveryNodeAction,
+  NodeValidatorAction,
+  SetNode,
+} from "./@types/contracts";
+import { AbstractTasteTest } from "./lib/classes/AbstractTasteTest.class";
+import {
+  FOLDING_FEE_ADA,
+  MIN_COMMITMENT_ADA,
+  NODE_ADA,
+  SETNODE_PREFIX,
+  TIME_TOLERANCE_MS,
+} from "./lib/contants";
+import { findCoveringNode } from "./utils";
+
+/**
+ * Object arguments for completing a transaction.
+ */
+interface ITasteTestCompleteTxArgs {
+  tx: Tx;
+  referralFee?: ITxBuilderReferralFee;
+  datum?: string;
+  complete?: boolean;
+}
+
+export class TasteTest implements AbstractTasteTest {
+  constructor(public lucid: Lucid) {}
+
+  public async deposit(
+    args: IDepositArgs
+  ): Promise<ITxBuilder<Tx, Datum | undefined>> {
+    const walletUtxos = await this.lucid.wallet.getUtxos();
+    const walletAddress = await this.lucid.wallet.address();
+
+    if (!walletUtxos.length) {
+      throw new Error("No available UTXOs found in wallet.");
+    }
+
+    if (!args.scripts) {
+      throw new Error(
+        "Did not receive a reference script UTXO or raw PlutusV2 CBOR script for the validator."
+      );
+    }
+
+    let nodeValidator: SpendingValidator;
+    if ("txHash" in args.scripts.validator) {
+      const script = (
+        await this.lucid.provider.getUtxosByOutRef([args.scripts.validator])
+      )?.[0];
```

Comment:
Is it always the first element?

Reply:
Yes, there's only ever one UTXO.

Reply:
since it's always the first element, we should probably assert that; also, it'd be great to be able to pass these in rather than fetching them from the provider.